### PR TITLE
Tagbox: Handle list not rendering immediately when setting tag values [T1230995]

### PR DESCRIPTION
**Pull request by nikkithelegendarypokemonster**

### CODE REVIEW

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 1]
```diff
@@ -5199,7 +5308,6 @@ QUnit.module('the \'fieldTemplate\' option', moduleSetup, () => {
     });
 });
 
-
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js:5202]: Usually between modules double line-breaks are used. It is not necessary to leave just one, especially since this change is not related to fixing the bug.

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 2]
```diff
@@ -4797,6 +4797,115 @@ QUnit.module('the \'acceptCustomValue\' option', moduleSetup, () => {
...
+            });
+            const instance = $element.dxTagBox('instance');
+            const $input = $element.find(`.${TEXTEDITOR_INPUT_CLASS}`);
+            const keyboard = keyboardMock($input, true);
+            keyboard
+                .focus()
+                .type(customValue)
+                .press('enter');
+
+            const isValueExist = instance.option('items').includes(customValue);
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: It's better to check the actual values of items rather than the presence of a new item in the list. In the previous test too.
```
assert.strictEqual(this.option('items'), expectedItems, 'Custom value is included in tagbox items');
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 3]
```diff
@@ -4797,6 +4797,115 @@ QUnit.module('the \'acceptCustomValue\' option', moduleSetup, () => {
...
+                    .type('test 1')
+                    .press('enter');
+
+                assert.ok(true, 'Error not encountered');
+            } catch(error) {
+                assert.ok(false, `Error encountered is: ${error.message}`);
+            }
+        });
+
+        QUnit.test(`if ${deferRendering} should display right selected items [T1230995]`, function(assert) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: The caption is nor informative: `if true should....`

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 4]
```diff
@@ -4797,6 +4797,115 @@ QUnit.module('the \'acceptCustomValue\' option', moduleSetup, () => {
...
+                        component.option('items', currentItems);
+                    }
+                    args.customItem = newValue;
+                },
+                deferRendering
+            });
+            const instance = $element.dxTagBox('instance');
+            const $input = $element.find(`.${TEXTEDITOR_INPUT_CLASS}`);
+            const keyboard = keyboardMock($input, true);
+            keyboard
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
            keyboardMock($input, true)
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 5]
```diff
@@ -4797,6 +4797,115 @@ QUnit.module('the \'acceptCustomValue\' option', moduleSetup, () => {
...
+                .focus()
+                .type(firstCustomValue)
+                .press('enter')
+                .type(secondCustomValue)
+                .press('enter');
+
+
+            if(deferRendering) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: Why we should have this condition here?

File: packages/devextreme/js/__internal/ui/m_tag_box.ts [hunk 6]
```diff
@@ -1295,17 +1295,18 @@ const TagBox = (SelectBox as any).inherit({
       return;
     }
 
+    const isListRendered = !this.option('deferRendering');
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/__internal/ui/m_tag_box.ts]: It seems that judging by the deferRendering option is not reliable that list was rendered. What will happen if the select box was already opened and closed? Will this code work?

Comment (nikkithelegendarypokemonster) [reply packages/devextreme/js/__internal/ui/m_tag_box.ts]: ok i will try to find alternative to check if list is rendered

Comment (nikkithelegendarypokemonster) [reply packages/devextreme/js/__internal/ui/m_tag_box.ts]: [30d85eb](https://github.com/DevExpress/DevExtreme/pull/27341/commits/30d85eb8b885fb8cfad6063c4b073a2970efa847)

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 5 above)
Comment (nikkithelegendarypokemonster) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: [30d85eb](https://github.com/DevExpress/DevExtreme/pull/27341/commits/30d85eb8b885fb8cfad6063c4b073a2970efa847)

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 4 above)
Comment (nikkithelegendarypokemonster) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: [30d85eb](https://github.com/DevExpress/DevExtreme/pull/27341/commits/30d85eb8b885fb8cfad6063c4b073a2970efa847)

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 3 above)
Comment (nikkithelegendarypokemonster) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: [30d85eb](https://github.com/DevExpress/DevExtreme/pull/27341/commits/30d85eb8b885fb8cfad6063c4b073a2970efa847)

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 2 above)
Comment (nikkithelegendarypokemonster) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: [30d85eb](https://github.com/DevExpress/DevExtreme/pull/27341/commits/30d85eb8b885fb8cfad6063c4b073a2970efa847)

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by iBat — APPROVED** [review]
